pkg/resource/cluster/hooks.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cluster
+
+import (
+	"errors"
+
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+)
+
+var (
+	condMsgCurrentlyDeleting     = "cluster currently being deleted"
+	condMsgNoDeleteWhileUpdating = "cluster is being updated. cannot delete"
+)
+
+var (
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		errors.New("delete is in progress"),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+	requeueWaitWhileUpdating = ackrequeue.NeededAfter(
+		errors.New("update is in progress"),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+)
+
+const (
+	available    = "available"
+	deleting     = "deleting"
+	updating     = "updating"
+	createFailed = "create-failed"
```

Comment:
nit: Please append const names with `Status`. eg. `StatusAvailable`. This is to be consistent with other controllers that do the same (and provides some context around what they're describing)